geonode_mapstore_client/client/js/api/geonode/v2/index.js

```diff
@@ -423,7 +409,7 @@ export const getResourcesTotalCount = () => {
         LAYERS,
         MAPS,
         GEOSTORIES,
-        GEOAPPS
+        DASHBOARDS
     ];
     return axios.all(
         types.map((type) =>
```

Comment:
I think we should modify this request and remove the workaround to get the total count by requesting 1 items for each resource type with axios.all and instead use only one request with the resource_type endpoint:

/api/v2/resources/resource_types

This is the current response:

![image](https://user-images.githubusercontent.com/19175505/125251704-e1eae800-e2f7-11eb-9305-7f3738bf97df.png)

Does this new improvement on the dashboard include the count of separated geoapps type in the resource_type endpoint ?

Reply:
Technically yes, the endpoint will return the `GeoApp type` with their count.
The only issue (that I see right now) is that we use the resource_type of the resource base to get the GeoApp details.
That could be a problem since if we don't have any kind of GeoApp in our geonode instance (for example a GeoStory), they will not be returned in the JSON.
If you want to make some live tests, you can switch the geonode to this branch from my fork `geoapp_refactor` (it required database migrations, so be careful about it)

Reply:
@allyoucanmap by adding the settings with the app list here is what the endpoint `/api/v2/resources/resource_types` will return:
```json
{
    "resource_types": [
        {
            "name": "layer",
            "count": 0
        },
        {
            "name": "map",
            "count": 0
        },
        {
            "name": "document",
            "count": 0
        },
        {
            "name": "geostory",
            "count": 0
        },
        {
            "name": "dashboard",
            "count": 0
        }
    ]
}
```

Reply:
see this comment https://github.com/GeoNode/geonode-mapstore-client/pull/302#discussion_r667767148